tests/foreman/cli/test_computeresource_vmware.py

```diff
@@ -0,0 +1,112 @@
+# -*- encoding: utf-8 -*-
+"""
+:Requirement: Computeresource
+
+:CaseLevel: Acceptance
+
+:CaseComponent: CLI
+
+:TestType: Functional
+
+:CaseImportance: High
+
+:Upstream: No
+"""
+from fauxfactory import gen_string
+
+from robottelo.config import settings
+from robottelo.constants import FOREMAN_PROVIDERS, VMWARE_CONSTANTS
+from robottelo.cli.factory import (
+    make_compute_resource,
+    make_location,
+    make_org,
+)
+from robottelo.cli.org import Org
+from robottelo.decorators import (
+    run_only_on,
+    skip_if_not_set,
+    tier1,
+)
+from robottelo.test import CLITestCase
+
+
+class VMWareComputeResourceTestCase(CLITestCase):
+    """VMware ComputeResource CLI tests."""
+
+    @classmethod
+    @skip_if_not_set('vmware')
+    def setUpClass(cls):
+        super(VMWareComputeResourceTestCase, cls).setUpClass()
+        cls.org = make_org()
+        cls.loc = make_location()
+        Org.add_location({'id': cls.org['id'], 'location-id': cls.loc['id']})
+        cls.vmware_server = settings.vmware.vcenter
+        cls.vmware_password = settings.vmware.password
+        cls.vmware_username = settings.vmware.username
+        cls.vmware_datacenter = settings.vmware.datacenter
+        cls.vmware_img_name = settings.vmware.image_name
+        cls.vmware_img_arch = settings.vmware.image_arch
+        cls.vmware_img_os = settings.vmware.image_os
+        cls.vmware_img_user = settings.vmware.image_username
+        cls.vmware_img_pass = settings.vmware.image_password
+        cls.vmware_vm_name = settings.vmware.vm_name
+        cls.current_interface = (
+            VMWARE_CONSTANTS.get(
+                'network_interfaces') % settings.vlan_networking.bridge
+        )
+
+    @tier1
+    @run_only_on('sat')
+    def test_positive_create_with_name(self):
+        """Create VMware Compute Resource
+
+        :id: 837c2091-609d-4a3a-8fcd-4cf9f4d0ba7b
+
+        :expectedresults: VMware Compute resource is created
+
+        :BZ: 1387917
+
+        :Caseautomation: Automated
+
+        :CaseImportance: Critical
+        """
+        cr_name = gen_string('alpha')
+        vmware_cr = make_compute_resource({
+            'name': cr_name,
+            'provider': FOREMAN_PROVIDERS['vmware'],
+            'server': self.vmware_server,
+            'user': self.vmware_username,
+            'password': self.vmware_password,
+            'datacenter': self.vmware_datacenter
+        })
+        self.assertEquals(vmware_cr['name'], cr_name)
+
+    @tier1
+    @run_only_on('sat')
+    def test_positive_create_with_server(self):
```

Comment:
I would say that second test is duplication to the first one or at least second one should focus on organization and location which is actual difference to first case.

Reply:
@oshtaier that was a problem as I have to make sure that the server field is saved correctly, and as we have no tests vmware cr for cli we have to make sure that a test with name exist, so the first test is a simple create with name to make sure that we can create a vmware cr,  the second one with org and loc and assert on server field saved

Reply:
Yeah, I can read the code=) So just make three cases ('test_positive_create_with_server`), (`test_positive_create_with_org`) and (`test_positive_create_with_server`). And check server + name in the first